rm-datasource/src/main/java/org/apache/seata/rm/datasource/sql/struct/TableMetaCacheFactory.java

```diff
@@ -142,7 +161,15 @@ static class TableMetaRefreshHolder {
             });
         }
 
-
-
+        /**
+         * Helper method to determine if the exception is caused by the data source being closed.
+         *
+         * @param ex the SQLException to check
+         * @return true if the exception indicates the data source is closed; false otherwise
+         */
+        private boolean isDataSourceClosedException(SQLException ex) {
```

Comment:
If it's just the connection being closed, can the two be distinguished?

Reply:
use sqlState 08006 and exception message keywords to distinguish